roles/ci_gen_kustomize_values/tasks/olm_subscriptions_overlay.yml

```diff
@@ -0,0 +1,57 @@
+---
+# Copyright Red Hat, Inc.
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+# Description: This playbook generates the kustomize file that
+# retrieves the appropriate overlay based on
+# cifmw_ci_gen_kustomize_values_deployment_version. It allows for
+# specifying the deployment version when multiple versions are
+# available in the OLM catalog. This is particularly useful for
+# testing updates but can also be used to deploy any version from the
+# OLM catalog.
+
+- name: Set the right overlay for the subscriptions
+  ansible.builtin.set_fact:
+    _cifmw_update_deployment_version_dir: >-
+      {{
+      cifmw_ci_gen_kustomize_values_deployment_version
+      if cifmw_ci_gen_kustomize_values_deployment_version in ['v1.0.3', 'v1.0.6']
+      else
+      'default'
+      }}
+
+- name: Point to the right overlay for OLM when deploying old version
+  vars:
+    _kustomization:
+      components:
+        - "../../../lib/olm-deps"
```

Comment:
nit: wondering, if here we could add absolute path.
I know that in other roles it is done same, so I will just leave a comment for me in the future ;)
I'm trying to work on local deployment for some test and that might be problematic.

Reply:
Yes this is  a copy of the original [examples/common/olm/kustomization.yaml](https://github.com/openstack-k8s-operators/architecture/blob/main/examples/common/olm/kustomization.yaml) where this definition is taken from.

Furthermore that code will only run for `v1.0.3` and `v1.0.6`.  Then we will directly using the new file from [this PR](https://github.com/openstack-k8s-operators/architecture/pull/537/files#diff-cada4dbe2af4cf7df05739d297c27b10d50e8817176838539480cf9bac95efe5) where definition is set.

Overall I think this should stay as it is.  Local testing using `kubectl kustomize` on that directory worked fine in my testing.